src/hera/workflows/_runner/script_annotations_util.py

```diff
@@ -46,6 +46,9 @@ def _get_dumper_function(destination: Union[Parameter, Artifact]) -> Callable:
     if isinstance(destination, Parameter) and destination.dumps is not None:
         return destination.dumps
 
+    if isinstance(destination, Artifact) and destination.dumps is not None:
+        return destination.dumps
+
     return serialize
```

Comment:
I think we can simplify this?
```suggestion
    return destination.dumps or serialize
```

Reply:
Just pushed the `dump` to bytes change which needed this code layout